diff --git a/src/webserver/webserver.cpp b/src/webserver/webserver.cpp
index 648693a..d8c2be6 100644
--- a/src/webserver/webserver.cpp
+++ b/src/webserver/webserver.cpp
@@ -735,8 +735,16 @@ void web_server_callback(MHDTransaction *mhdt)
 			if (fd < 0) {
 				http_SendStatusResponse(mhdt, HTTP_FORBIDDEN);
 			} else {
+#if MHD_VERSION <= 0x00093700
+				// Not sure exactly at_offset64 appeared, but 0.9.37
+				// did not have it
+				mhdt->response = MHD_create_response_from_fd_at_offset(
+					RespInstr.ReadSendSize, fd,
+					static_cast<off_t>(RespInstr.offset));
+#else
 				mhdt->response = MHD_create_response_from_fd_at_offset64(
 					RespInstr.ReadSendSize, fd, RespInstr.offset);
+#endif
 				mhdt->httpstatus = 200;
 			}
 		}
